sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -550,19 +551,55 @@ var sloSLITriggerConditionSchema = map[string]*schema.Schema{
 
 var sloBurnRateTriggerConditionSchema = map[string]*schema.Schema{
 	"critical": nestedWithAtleastOneOfKeys(true, schemaMap{
-		"time_range": &timeRangeSchema,
+		"time_range": getSloBurnRateTimeRangeSchema("critical"),
 		"burn_rate_threshold": {
-			Type:         schema.TypeFloat,
-			Required:     true,
-			ValidateFunc: validation.FloatAtLeast(0),
+			Type:          schema.TypeFloat,
+			Optional:      true,
+			ValidateFunc:  validation.FloatAtLeast(0),
+			RequiredWith:  []string{"trigger_conditions.0.slo_burn_rate_condition.0.critical.0.time_range"},
+			ConflictsWith: []string{"trigger_conditions.0.slo_burn_rate_condition.0.critical.0.burn_rate"},
+		},
+		"burn_rate": {
+			Optional: true,
+			Type:     schema.TypeList,
+			ConflictsWith: []string{"trigger_conditions.0.slo_burn_rate_condition.0.critical.0.burn_rate_threshold",
+				"trigger_conditions.0.slo_burn_rate_condition.0.critical.0.time_range"},
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"burn_rate_threshold": {
+						Type:         schema.TypeFloat,
+						Required:     true,
+						ValidateFunc: validation.FloatAtLeast(0),
+					},
+					"time_range": &timeRangeSchema,
+				},
+			},
 		},
 	}, sloBurnRateConditionCriticalOrWarningAtleastOneKeys),
 	"warning": nestedWithAtleastOneOfKeys(true, schemaMap{
-		"time_range": &timeRangeSchema,
+		"time_range": getSloBurnRateTimeRangeSchema("warning"),
 		"burn_rate_threshold": {
-			Type:         schema.TypeFloat,
-			Required:     true,
-			ValidateFunc: validation.FloatAtLeast(0),
+			Type:          schema.TypeFloat,
```

Comment:
why not also have a similar method for `getSloBurnRateThresholdSchema`

Reply:
updated, [commit](https://github.com/SumoLogic/terraform-provider-sumologic/pull/499/commits/88c52da73c41464be2c8873369b0b2ba207afa0e#diff-b749eb8ae5017ef0ed2545e094d7e38321f015a91b335265140abcf75fb61365R629:~:text=func-,getSloBurnRateThresholdSchema,-(triggerType%20string)